test/integration/spec/rtcpeerconnection.js

```diff
@@ -786,43 +739,140 @@ function makeStream(constraints) {
   return new Promise((resolve, reject) => getUserMedia(resolve, reject));
 }
 
-function testAddStream() {
+function testAddTrack() {
   var test;
   var stream;
+  var trackToAdd;
 
-  before(() => {
-    return makeStream().then(_stream => stream = _stream);
+  before(async () => {
+    stream = await makeStream();
+    trackToAdd = stream.getTracks()[0];
   });
 
-  beforeEach(() => {
-    return makeTest().then(_test => test = _test);
+  beforeEach(async () => {
+    test = await makeTest();
   });
 
-  // NOTE(mroberts): See the comment in lib/webrtc/rtcpeerconnection/firefox.js
-  // for an explanation as to why we test this API the way we do.
-  it('should add each of the MediaStream\'s MediaStreamTracks to the RTCPeerConnection', () => {
-    test.peerConnection.addStream(stream);
-    if (isFirefox) {
-      const expectedTracks = stream.getTracks();
-      const actualTracks = test.peerConnection.getLocalStreams()[0].getTracks();
-      assertMediaStreamTracksEqual(actualTracks, expectedTracks);
-      return;
-    }
-    assert.equal(test.peerConnection.getLocalStreams()[0], stream);
-  });
+  [
+    [
+      'when the RTCPeerConnection is closed',
+      () => test.peerConnection.close()
+    ],
+    [
+      'when the MediaStreamTrack is already added to the RTCPeerConnection',
+      () => test.peerConnection.addTrack(trackToAdd, stream)
+    ]
+  ].forEach(([ scenario, setup ], i) => {
+    context(scenario, () => {
+      var exception;
 
-  context('when adding a stream that is already added', () => {
-    it('should not throw an exception', () => {
-      test.peerConnection.addStream(stream);
-      assert.doesNotThrow(() => test.peerConnection.addStream(stream));
+      beforeEach(() => {
+        setup();
+        try {
+          test.peerConnection.addTrack(trackToAdd, stream);
+        } catch (e) {
+          exception = e;
+        }
+      });
+
+      it('should throw', () => {
+        assert(exception);
+      });
+
+      if (i === 0) {
```

Comment:
Rather than depend on `i === 0`, i.e., the order of the test scenarios above, we should be explicit as to whether or not we expect it to fail.